pkg/pillar/cmd/verifier/verifier.go

```diff
@@ -742,29 +741,38 @@ func markObjectAsVerifying(ctx *verifierContext,
 func verifyObjectSha(ctx *verifierContext, config *types.VerifyImageConfig,
 	status *types.VerifyImageStatus) bool {
 
-	if status.IsContainer {
-		log.Infof("verifyObjectSha: Container image %s. Returning success",
-			config.Name)
-		return true
-	}
 	_, verifierFilename, _ := status.ImageDownloadFilenames()
 	log.Infof("Verifying URL %s file %s\n", config.Name, verifierFilename)
 
-	imageHash, err := computeShaFile(verifierFilename)
+	imageHashB, err := computeShaFileByType(verifierFilename, status.IsContainer)
 	if err != nil {
 		cerr := fmt.Sprintf("%v", err)
 		updateVerifyErrStatus(ctx, status, cerr)
 		log.Errorf("verifyObjectSha %s failed %s\n",
 			config.Name, cerr)
 		return false
 	}
-
-	got := fmt.Sprintf("%x", imageHash)
-	if got != strings.ToLower(config.ImageSha256) {
-		log.Errorf("computed   %s\n", got)
-		log.Errorf("configured %s\n", strings.ToLower(config.ImageSha256))
+	log.Infof("internal hash consistency validated for URL %s file %s", config.Name, verifierFilename)
```

Comment:
By "internal" do you mean from the root down the layers?

Reply:
Sort of. The verifier will do two things:

1. Always take the image manifest inside the tar file, and verify that the layers and config embedded in the same file have hashes that match their file names and the hashes provided in the manifest. This is what I meant by "internal".
2. If a hash is provided by the config, check that the image manifest itself (the root) has a hash that matches that which is provided by the config.